src/main/webapp/js/components/fields/field-config-modal.jsx

```diff
@@ -0,0 +1,260 @@
+import React from 'react';
+import Modal from 'react-modal';
+import PropTypes from 'prop-types';
+import { Form } from 'react-final-form';
+import ConfirmModal from '../comfirm-modal';
+
+import renderFormField from '../form/form-utils';
+
+Modal.setAppElement(document.getElementById('root'));
+
+const customStyles = {
+  content: {
+    top: '62%',
+    left: '50%',
+    right: 'auto',
+    bottom: 'auto',
+    marginRight: '-50%',
+    transform: 'translate(-50%, -50%)',
+  },
+  overlay: {
+    overflow: 'auto',
+  },
+};
+
+const FIELD_TYPE_OPTIONS = [
+  { label: '', value: null },
+  { label: 'Text', value: 'TEXT' },
+  { label: 'Long text', value: 'LONG_TEXT' },
+  { label: 'Integer', value: 'INTEGER' },
+  { label: 'Float', value: 'FLOAT' },
+  { label: 'Boolean', value: 'BOOLEAN' },
+  { label: 'Data', value: 'DATE' },
+  { label: 'Date time', value: 'DATE_TIME' },
+  { label: 'Enum', value: 'ENUM' },
+  { label: 'Relation', value: 'RELATION' },
+  { label: 'Collection', value: 'COLLECTION' },
+  { label: 'Vaccination date', value: 'VACCINATION_DATE' },
+];
+
+const ENTITY_TYPE_OPTION = [
+  { label: '', value: null },
+  { label: 'Vaccinee', value: 'VACCINEE' },
+  { label: 'Visit', value: 'VISIT' },
+  { label: 'Site', value: 'SITE' },
+  { label: 'Group', value: 'GROUP' },
+  { label: 'Person', value: 'PERSON' },
+  { label: 'Language', value: 'LANGUAGE' },
+  { label: 'Visit type', value: 'VISIT_TYPE' },
+];
+
+class FieldConfigModal extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      openConfirmModal: false,
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { item } = nextProps;
+    this.mapConfigToState(item);
+  }
+
+  onSubmit = (values) => {
+    console.log(values);
+    this.props.updateConfig(values);
+    this.props.toggleModal();
+  };
+
+  afterOpenModal = () => {
+    this.subtitle.style.color = '#f00';
+  };
+
+  hideConfirmModal = () => {
+    this.setState({ openConfirmModal: false });
+  };
+
+  hideConfigModal = () => {
+    this.hideConfirmModal();
+    this.props.toggleModal();
+  };
+
+  deleteConfig = () => {
+    this.props.deleteConfig(this.props.item, () => this.hideConfigModal());
+  };
+
+  displayFields = (item) => {
+    const {
+      base, editable, filterable, required, fieldType, entity,
+      displayName, name, format, relatedEntity, relatedField, openConfirmModal,
+    } = this.state;
+    return (
+      !!item && (
+      <div className="config-modal-form">
+        <Form
+          onSubmit={this.onSubmit}
+          initialValues={{
```

Comment:
you can just use this.props.item directly in here (form is not mutating the initialValues)